### ISSUE TITLE
Aria live overly used in components. 

### ISSUE THREAD

**Issue by colinhale1** [Esri/calcite-design-system#10704]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

During our recent accessibility audit, we found excessive use of the aria-live attribute in the calcite-button, calcite-tooltip, and calcite-modal components. This attribute is added in the shadow root, which we don't control.

For more context, here is the full issue report from the accessibility audit for one of the instances:

> Redundant "aria-live" attributes have been added to static (non-changing) containers throughout the page. Instances include:
> 
> Tooltips
> Resources dialog and buttons within the dialog.
> Analysis extent button
> Current map extent popover
> [User Impact]
> When markup is not implemented to spec, assistive tech can act unpredictably. In this instance, it could cause assistive tech to convey a lot of redundant information which their users would then need to parse, trying to find the information they actually need.
> 
> [Code Reference]
> <calcite-popover class="esriGECalcitePopover" aria-hidden="false" aria-live="polite" id="calcite-popover-577c8a09-26ff-0635-d6a7-b703768c8e80" role="dialog" (...)>(...)
> 
> [Steps to Reproduce]
> 
> Enable the screen reader.
> Navigate to "Resources" and activate it.
> Notice that all the content within the dialog is conveyed by the screen reader.
> Press Tab and notice that the close button is focused and conveyed three times, all the dialog's content is conveyed again and controls information is conveyed twice.
> Note
> [Recommendation]
> Ensure ARIA roles, states, and properties are valid. Many ARIA attributes can only be used on certain elements. Some can only be used in combination with other attributes. Additionally, if an attribute references an ID, that ID must be present in the page.
> To detect many of these errors automatically, use the Nu Html Checker: https://validator.w3.org/nu/
> 
> Content and controls that do not change nor update must not have aria-live="polite" this attribute should only be used for dynamically changing content that occurs in content placed before the point where the focus is placed or when changes in content need to be conveyed right after they occur.
> 
> [Compliant Code Example]
> <calcite-popover class="esriGECalcitePopover" style="z-index: 999; position: fixed; top: 0px; left: 0px; transform: translate(89px, 163px);" aria-hidden="false" id="calcite-popover-577c8a09-26ff-0635-d6a7-b703768c8e80" role="dialog" aria-labelledby="resources-dialog-title" (...)>

### Actual Behavior

Screen shot of the code pen calcite-tooltip example with the aria-live.
![Screenshot 2024-11-06 at 3 46 00 PM](https://github.com/user-attachments/assets/b3047390-c2d0-4e1e-aecb-d3954a0d5919)

### Expected Behavior

The attribute aria-live should be optional and not used by default.

### Reproduction Sample

https://codepen.io/pen?&editors=100

### Reproduction Steps

1. Open example
2. Inspect tooltip
3. Observe aria-live is added to component

### Reproduction Version

Current

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by geospatialem** [comment]
Spike to determine:
- Which components use `aria-live`
- Do all the components using `aria-live` need to depict dynamic information/updates?

**Comment by fdeters** [comment]
This issue came up for Oregon Metro in a recent accessibility audit, specifically with `calcite-action`'s aria-live indicator region. Even when we're not using the indicator at all, it shows up as a region when screen reader users are browsing, which is pretty confusing. 

If a separate issue would be helpful for our specific case with `region`s disrupting the document flow, let me know and I'll create one 👍

**Comment by jcfranco** [comment]
Related to ☝️, this was something that came up when porting the `accessible` test helper to browser mode:

> Expected the HTML found at $('calcite-combobox,#combobox-chip-8169bfa2-b469-6e27-a40c-da787066d3d3,calcite-action,.indicator-text[role="region"][aria-live="polite"]') to have no violations:
> 
> <div aria-live="polite" role="region" aria-labelledby="calcite-action-e6d46f72-470d-39b5-8b68-e091e9ed6a44-button" class="indicator-text"></div>
> 
> Received:
> 
> "Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)"
> 
> Fix any of the following:
>   The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable
> 
> You can find more information on this issue here:
> https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

Changing the [role for the indicator](https://github.com/Esri/calcite-design-system/pull/13361/changes#diff-7a4dfcc2686412d4036d13bd988a7e385281e7e4a2a7abb254b2804183edaa3e) mitigated the error, but needs to be properly reviewed. @geospatialem
